### PR TITLE
TaskError should report which task failed

### PR DESCRIPTION
**Pull request by younata**

### CODE REVIEW

**Comment by younata** [comment]
Carthage's edge of master is broken at the moment, using this to help divine the root cause. Thought it would be helpful for anyone in the future.

File: ReactiveTask/Errors.swift [hunk 1]
```diff
@@ -12,7 +12,7 @@ import ReactiveCocoa
 /// An error originating from ReactiveTask.
 public enum TaskError: ErrorType {
 	/// A shell task exited unsuccessfully.
-	case ShellTaskFailed(exitCode: Int32, standardError: String?)
+	case ShellTaskFailed(task: Task, exitCode: Int32, standardError: String?)
```
Comment (mdiep) [inline ReactiveTask/Errors.swift]: Can we remove the `task:` label from the `Task` argument? The meaning of the argument should be obvious, so I don't think the label is necessary.

**Comment by mdiep** [comment]
I like it. :+1: Just the one comment.

**Comment by younata** [comment]
Done.

**Comment by mdiep** [comment]
That was quick!

**Pull request merged**